internal/plugincommon/discovery/processor.go

```diff
@@ -204,16 +239,46 @@ func (cdp *ContractDiscoveryProcessor) Outcome(
 	rmnRemoteConsensus := consensus.GetConsensusMap(
 		cdp.lggr,
 		"rmnRemote",
-		map[cciptypes.ChainSelector][][]byte{cdp.dest: rmnRemoteAddrs},
+		map[cciptypes.ChainSelector][][]byte{cdp.dest: agg.rmnRemoteAddrs},
 		fDestChainThresh,
 	)
 	cdp.lggr.Infow("Determined consensus RMNRemote",
 		"rmnRemoteConsensus", rmnRemoteConsensus,
-		"rmnRemoteAddrs", rmnRemoteAddrs,
+		"rmnRemoteAddrs", agg.rmnRemoteAddrs,
 		"fDestChainThresh", fDestChainThresh,
 	)
 	contracts[consts.ContractNameRMNRemote] = rmnRemoteConsensus
 
+	// Router address consensus
+	routerConsensus := consensus.GetConsensusMap(
+		cdp.lggr,
+		"router",
+		map[cciptypes.ChainSelector][][]byte{cdp.dest: agg.routerAddrs},
+		fDestChainThresh,
+	)
+	cdp.lggr.Infow("Determined consensus Router",
+		"RouterConsensus", routerConsensus,
+		"RouterAddrs", agg.routerAddrs,
+		"fDestChainThresh", fDestChainThresh,
+	)
+	contracts[consts.ContractNameRouter] = routerConsensus
+
+	feeQuoterConsensus := consensus.GetConsensusMap(
+		cdp.lggr,
+		"fee quoter",
+		agg.feeQuoterAddrs,
+		fDestChainThresh,
+	)
+	cdp.lggr.Infow("Determined consensus fee quoter",
+		"feeQuoterConsensus", feeQuoterConsensus,
+		"feeQuoterAddrs", agg.feeQuoterAddrs,
+		"fDestChainThresh", fDestChainThresh,
+	)
+	if len(feeQuoterConsensus) == 0 {
```

Comment:
Why only this condition for feeQuoter and not the others?

Reply:
The others should always be present. But maybe thats even more of a reason to add add the warning.

Reply:
added.

Reply:
added others.